src/app/modules/main/wallet_section/add_account/controller.nim

```diff
@@ -0,0 +1,229 @@
+    var args = DerivedAddressesArgs(e)
+    self.delegate.onDerivedAddressesFromMnemonicFetched(args.derivedAddresses, args.error)
+  self.connectionIds.add(handlerId)
+
+  handlerId = self.events.onWithUUID(SIGNAL_DERIVED_ADDRESSES_FROM_NOT_IMPORTED_MNEMONIC_FETCHED) do(e:Args):
+    var args = DerivedAddressesFromNotImportedMnemonicArgs(e)
+    self.delegate.onAddressesFromNotImportedMnemonicFetched(args.derivations, args.error)
+  self.connectionIds.add(handlerId)
+
+  handlerId = self.events.onWithUUID(SIGNAL_WALLET_ACCOUNT_ADDRESS_DETAILS_FETCHED) do(e:Args):
+    var args = DerivedAddressesArgs(e)
+    if args.uniqueId != self.uniqueFetchingDetailsId:
+      return
+    self.delegate.onAddressDetailsFetched(args.derivedAddresses, args.error)
+  self.connectionIds.add(handlerId)
+
+proc setAuthenticatedKeyUid*(self: Controller, value: string) =
+  self.tmpAuthenticatedKeyUid = value
+
+proc getAuthenticatedKeyUid*(self: Controller): string =
+  return self.tmpAuthenticatedKeyUid
+
+proc setPin*(self: Controller, value: string) =
+  self.tmpPin = value
+
+proc getPin*(self: Controller): string =
+  return self.tmpPin
+
+proc setPassword*(self: Controller, value: string) =
+  self.tmpPassword = value
+
+proc getPassword*(self: Controller): string =
+  return self.tmpPassword
+
+proc setSeedPhrase*(self: Controller, value: string) =
+  self.tmpSeedPhrase = value
+
+proc getSeedPhrase*(self: Controller): string =
+  return self.tmpSeedPhrase
+
+proc closeAddAccountPopup*(self: Controller) =
+  self.delegate.closeAddAccountPopup()
+
+proc getWalletAccounts*(self: Controller): seq[WalletAccountDto] =
+  return self.walletAccountService.fetchAccounts()
+
+proc getAllMigratedKeyPairs*(self: Controller): seq[KeyPairDto] =
+  return self.walletAccountService.getAllMigratedKeyPairs()
+
+proc addAccount*(self: Controller) =
+  self.delegate.addAccount()
+
+proc authenticateOrigin*(self: Controller, keyUid = "") =
+  let data = SharedKeycarModuleAuthenticationArgs(uniqueIdentifier: UNIQUE_WALLET_SECTION_ADD_ACCOUNTS_MODULE_IDENTIFIER,
+    keyUid: keyUid)
+  self.events.emit(SIGNAL_SHARED_KEYCARD_MODULE_AUTHENTICATE_USER, data)
+
+method fetchDerivedAddresses*(self: Controller, derivedFrom: string, paths: seq[string])=
+  var hashPassword = true
+  if self.getPin().len > 0:
+    hashPassword = false
+  self.walletAccountService.fetchDerivedAddresses(self.getPassword(), derivedFrom, paths, hashPassword)
+
+proc getRandomMnemonic*(self: Controller): string =
+  return self.walletAccountService.getRandomMnemonic()
+
+proc fetchDetailsForAddresses*(self: Controller, addresses: seq[string]) =
+  self.uniqueFetchingDetailsId = $now().toTime().toUnix()
+  self.walletAccountService.fetchDetailsForAddresses(self.uniqueFetchingDetailsId, addresses)
+
+proc addWalletAccount*(self: Controller, createKeystoreFile, doPasswordHashing: bool, name, keyPairName, address, path: string, 
```

Comment:
@stefandunca `Module`, where we usually keep the logic for a module, decides based on some logic whether we need to create a keystore file or not. Logic is that if a keypair we're adding an account for is migrated to a keycard we don't create keystore file, otherwise we create it.